src/buffer/out/LineRendition.hpp

```diff
@@ -35,6 +42,13 @@ constexpr til::point ScreenToBufferLine(const til::point& line, const LineRendit
     return { line.x >> scale, line.y };
 }
 
+constexpr til::rect BufferToScreenLine(const til::rect& line, const LineRendition lineRendition)
+{
+    // Use shift left to quickly multiply the Left and Right by 2 for double width lines.
+    const auto scale = lineRendition == LineRendition::SingleWidth ? 0 : 1;
+    return { line.left << scale, line.top, (line.right << scale) + scale, line.bottom };
+}

```

Comment:
I think this may also be off by one when dealing with exclusive coordinates. For example, if the right buffer coordinate is 4 exclusive (3 inclusive), then the right screen coordinate should be 8 exclusive (7 inclusive). The current formula works for inclusive coordinates (`(3 << 1) + 1 = 7`), but not exclusive coordinates (`(4 << 1) + 1 = 9`, but we want 8).